### PR TITLE
Fix several aspects of error management to throw errors only when they actually matter

### PR DESCRIPTION
**Pull request by Soulou**

### CODE REVIEW

File: web/networks_controller_connect.go [hunk 1]
```diff
@@ -76,7 +79,15 @@ func (c NetworksController) Connect(w http.ResponseWriter, r *http.Request, urlp
...
 		err := netutils.ForwardConnection(ctx, socket, localEndpoint.TargetNetnsPath, ip, port)
-		if err != nil {
+		// It happens that the target from the network connection (ip:port) is not
+		// actually bound in the network namespace, in this case a standard
+		// connection refused error is sent, this should not be an error, the
+		// connection to the SAND client should just be stopped normally
+		if err, ok := errors.Cause(err).(*net.OpError); ok {
+			if err, ok := err.Err.(*os.SyscallError); ok && err.Err == syscall.ECONNREFUSED {
+				log.WithError(err).Infof("local endpoint %v not binding port", localEndpoint)
+			}
```
Comment (EtienneM) [inline web/networks_controller_connect.go]: Aren't we missing an error to display in the `else` case here?

File: web/networks_controller_connect.go [hunk 2]
```diff
@@ -76,7 +79,15 @@ func (c NetworksController) Connect(w http.ResponseWriter, r *http.Request, urlp
...
 	if localEndpoint.ID != "" {
 		err := netutils.ForwardConnection(ctx, socket, localEndpoint.TargetNetnsPath, ip, port)
-		if err != nil {
+		// It happens that the target from the network connection (ip:port) is not
+		// actually bound in the network namespace, in this case a standard
+		// connection refused error is sent, this should not be an error, the
+		// connection to the SAND client should just be stopped normally
+		if err, ok := errors.Cause(err).(*net.OpError); ok {
+			if err, ok := err.Err.(*os.SyscallError); ok && err.Err == syscall.ECONNREFUSED {
+				log.WithError(err).Infof("local endpoint %v not binding port", localEndpoint)
```
Comment (EtienneM) [inline web/networks_controller_connect.go]: Don't you want to return here?

Comment (Soulou) [reply web/networks_controller_connect.go]: There is a `return nil`  line 93, it gets in it

File: web/networks_controller_connect.go (hunk 1 above)
Comment (Soulou) [reply web/networks_controller_connect.go]: Indeed we are, handling it

**Pull request merged**